File: pob-sdk/pobeye2/examples/pob_bot_demo/pad.c
/****************************************/
/* POB-TECHNOLOGY						*/
/* file pad.c							*/	
/* 16-02-2006							*/
/* Demo for PobBoTDemo					*/
/* author Pierre Seguin					*/ 
/* e-mail :pseguin@pob-technology.com	*/
/****************************************/
#include "POB-EYE.h"
#include "pad.h"





/**************** externs variables defined in main.c***************/
extern UInt8 X_Cursor;
extern UInt8 Y_Cursor;
/*******************************************************************/


//function to get the position of the joystick and manage the X-Y coordinates of the cursor
//return 1 if the cursor has moved, else return 0
UInt8 GetPad(void)
{
	UInt8 exit=0;
	if (GetPortAnalog(RIGHT_LEFT_AXE)>200)		//test the right_left axe
	{		
		Y_Cursor-=2;	//move the cursor to the right
		exit=1;
	}

	else if (GetPortAnalog(RIGHT_LEFT_AXE)<50)  //test the right_left axe
	{		
		Y_Cursor+=2;	//move the cursor to the left
		exit=1;
	}

	if (GetPortAnalog(UP_DOWN_AXE)>200)			//test the up_down axe
	{		
		X_Cursor+=2;	//move the cursor to the top
		exit=1;
	}

	else if (GetPortAnalog(UP_DOWN_AXE)<50) //test the up down axe
	{		
		X_Cursor-=2;	//move the cursor to the bottom
		exit=1;
	}

	return exit;
}




//function to know if the button is pressed
// return 1 if it's true, else return 0
UInt8 GetButton (void)
{
	if (GetPortAnalog(BUTTON)<100)
	{		
		return 1;
	}
	else
	{		
		return 0;
	}
}

File: pob-sdk/pobeye2-java/examples/pob-bot-demo/Pad.java
// Copyrith POB-Technology, 2006. All rights reserved.

import com.pobtechnology.pobeye.* ;

/**
 * Joystick and cursor management.
 *
 * @author Baptiste Burles & Pierre Séguin POB-Technology
 */
class Pad
{
	/**
	 * x position cursor
	 */
	public static int X;
	/**
	 * y position cursor
	 */
	public static int Y;	


	/**
	 * Init cursor
	 */	
	public static void init()
	{
		Pad.X=100;
		Pad.Y=32;
	}
	
	/**
	 * Get new position of joystick 
	 *
	 * @return true if joystick has moved
	 */
	public static boolean get()
	{
		boolean result=false;	

		int up_down = PobProto.getPortAnalog(UP_DOWN_AXE);
		if ( up_down >200)
		{
			X+=2;
			result=true;
		}
		else if(up_down<50)
		{
			X-=2;
			result=true;
		}
		
		int right_left = PobProto.getPortAnalog(RIGHT_LEFT_AXE);
		if ( right_left > 200)
		{
			Y-=2;
			result=true;
		}
		else if ( right_left < 50 )
		{
			Y+=2;
			result=true;
		}
		
		if( Pad.Y > 60 )
			Pad.Y = 60 ;
		if( Pad.X > 125 )
			Pad.X = 125 ;
		if( Pad.Y < 0 )
			Pad.Y = 0 ;
		if( Pad.X < 0 )
			Pad.X = 0;
			
		return result ;
	}

	/**
	 * Get button pressed 
	 *
	 * @return true if button pressed on joystick
	 */
	public static boolean getButton()
	{
	
		if( PobProto.getPortAnalog(BUTTON)<100)
		{
			return true;
		}
		else
		{		
			return false;
		}
	}
	
	private Pad()
	{
	}
	
	// internal order for get joystick position and button pressed
	private final static int RIGHT_LEFT_AXE	=0;
	private final static int UP_DOWN_AXE	=1;
	private final static int BUTTON	=4;
	
}